### PR TITLE
ci(android-java): add BrowserStack integration tests

### PR DESCRIPTION
**Pull request by teodorciuraru**
## Summary
• Add production-ready Android Java CI/CD pipeline with BrowserStack integration following Kotlin workflow patterns
• Implement comprehensive testing strategy with graceful failure handling for all edge cases
• **Success test**: Verifies seeded documents appear correctly in UI with proper synchronization
• **Empty environment test**: Gracefully handles missing GITHUB_TEST_DOC_ID by failing fast with clear messaging  
• **Non-existing document test**: Properly handles document-not-found scenarios with appropriate timeouts
• Use inverted timestamps for predictable document ordering ensuring latest items appear at top
• Test across real device matrix: Google Pixel 8, Samsung Galaxy S23, Google Pixel 6, OnePlus 9

## Technical Implementation
• **CI Architecture**: Follows proven 3-stage pattern: `lint` → `build-and-test` → `browserstack-test`
• **BrowserStack Integration**: Single build upload with comprehensive device matrix testing
• **Test Strategy**: Each test method handles its expected conditions gracefully - no unexpected failures
• **Environment Handling**: Robust environment variable validation with clear error messaging
• **UI Testing**: Espresso-based tests with proper wait strategies and view matching

## Test Plan
- [x] Local emulator testing validates all test scenarios handle expected conditions properly
- [x] Empty environment variable test passes immediately by detecting missing configuration
- [x] Non-existing document test passes by gracefully handling NoMatchingViewException within timeout
- [x] Success test validates end-to-end document sync and UI display (requires CI seeded document)
- [x] CI pipeline mirrors successful Kotlin implementation patterns
- [x] BrowserStack device matrix covers major Android versions and manufacturers

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: android-java/app/src/main/java/com/example/dittotasks/MainActivity.java [hunk 1]
```diff
@@ -84,14 +90,20 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
...
     }
 
-    void initDitto() {
-        requestPermissions();
 
+    void initDitto() {
+        // Skip permission requests during testing to avoid permission dialogs
+        if (!isInstrumentationTest()) {
+            requestPermissions();
+        }
```
Comment (teodorciuraru) [inline android-java/app/src/main/java/com/example/dittotasks/MainActivity.java:103]: Hmm, this logic isn't ideal for the Main app. I'll look into moving it later.
